File: android/src/main/java/expo/modules/blekit/BlekitModule.kt
package expo.modules.blekit

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BlekitModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("Blekit")

    Events("onChange")

    Function("hello") {
      "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { value: String ->
      sendEvent("onChange", mapOf(
        "value" to value
      ))
    }
  }
}

File: ios/BlekitModule.swift
import ExpoModulesCore

public class BlekitModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Blekit")

    Events("onChange")

    Function("hello") {
      return "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { (value: String) in
      self.sendEvent("onChange", [
        "value": value
      ])
    }
  }
}
